extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -669,25 +666,12 @@ public String getEarliestSequenceNumber(StreamPartition<String> partition)
     return wrapExceptions(
         () -> {
           final Set<String> retVal = new HashSet<>();
-          DescribeStreamRequest request = new DescribeStreamRequest();
+          ListShardsRequest request = new ListShardsRequest();
           request.setStreamName(stream);
-
-          while (request != null) {
-            final DescribeStreamResult result = kinesis.describeStream(request);
-            final StreamDescription streamDescription = result.getStreamDescription();
-            final List<Shard> shards = streamDescription.getShards();
-
-            for (Shard shard : shards) {
-              retVal.add(shard.getShardId());
-            }
-
-            if (streamDescription.isHasMoreShards()) {
-              request.setExclusiveStartShardId(Iterables.getLast(shards).getShardId());
-            } else {
-              request = null;
-            }
+          List<Shard> shards = kinesis.listShards(request).getShards();
```

Comment:
doubt : is this call paginated? (since the original API call is paginated with 1k shards). and also the `ListShardResult` also contains a nextToken. If it is paginated, I think we should get all the results

Reply:
Thanks for the scrutiny. It does seem like there is a limit of 1k (despite over 1100 shards being returned during my local testing).
I'll make the changes